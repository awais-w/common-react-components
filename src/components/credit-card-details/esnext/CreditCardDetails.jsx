import React from 'react';
import bem from '@argos/utils/esnext/bem';
import moment from 'moment';
import propTypes from './CreditCardDetails.propTypes';

import './CreditCardDetails.scss';

const cn = bem({ prefix: 'cmc-', block: 'credit-card-details' });

function displayDate(startDate, expiryDate) {
  if (startDate) {
    if (expiryDate) {
      return `${startDate} - ${expiryDate}`;
    }
    return `${startDate} - No expiration`;
  }
  if (expiryDate) {
    return `Expiration date - ${expiryDate}`;
  }
  return `No expiration provided`;
}

const CreditCardDetails = ({ type, number, startDate, expiryDate, name }) => {
  const formattedStartDate = startDate && moment(startDate).format('MM/YY');
  const formattedExpiryDate = expiryDate && moment(expiryDate).format('MM/YY');
  const formattedDate = displayDate(formattedStartDate, formattedExpiryDate);

  return (
    <div className={cn()}>
      <i className='cmc-icon' role='icon'>
        credit_card
      </i>
      {type && <span>{type}</span>}
      <span>{number}</span>
      {formattedDate && <span>{formattedDate}</span>}
      {name && <span>{name}</span>}
    </div>
  );
};

CreditCardDetails.propTypes = propTypes;

export default CreditCardDetails;
