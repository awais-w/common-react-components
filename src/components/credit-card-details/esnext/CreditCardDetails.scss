@import '@cmc/common-styles/src/scss/_variables.scss';

.cmc-credit-card-details {
  @include font-standard;
  display: flex;
  align-items: center;
  i {
    padding-right: $half;
  }
  span {
    padding-right: $base;
  }
}
