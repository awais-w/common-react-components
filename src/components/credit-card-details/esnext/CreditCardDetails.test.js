import React from 'react';
import { shallow } from 'enzyme';
import CreditCardDetails from './CreditCardDetails';

const allProps = {
  type: 'Argos Card',
  number: 'XXXX-XXXX-XXXX-1234',
  startDate: '2019-01-14T05:32:45Z',
  expiryDate: '2099-05-14T05:32:45Z',
  name: 'Mr Test Person',
};

describe('CreditCardDetails', () => {
  const component = shallow(<CreditCardDetails {...allProps} />);

  it('should match all props snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('should match required props snapshot', () => {
    const onlyRequiredProps = {
      number: 'XXXX-XXXX-XXXX-1234',
    };
    const updatedComponent = shallow(<CreditCardDetails {...onlyRequiredProps} />);
    expect(updatedComponent).toMatchSnapshot();
  });

  it('has the credit card icon', () => {
    expect(component.find('.cmc-icon').exists()).toBeTruthy();
  });

  describe('date options', () => {
    it('should render the correct date', () => {
      expect(component.find('span[children="01/19 - 05/99"]').exists()).toBeTruthy();
    });

    it('should render only startDate with no expiryDate', () => {
      const updatedProps = { ...allProps, expiryDate: '' };
      const noExpiryComponent = shallow(<CreditCardDetails {...updatedProps} />);
      expect(noExpiryComponent.find('span[children="01/19 - No expiration"]').exists()).toBeTruthy();
    });

    it('should render only expiryDate with no startDate', () => {
      const updatedProps = { ...allProps, startDate: '' };
      const noStartComponent = shallow(<CreditCardDetails {...updatedProps} />);
      expect(noStartComponent.find('span[children="Expiration date - 05/99"]').exists()).toBeTruthy();
    });

    it('should render only the fallback message', () => {
      const updatedProps = { ...allProps, startDate: '', expiryDate: '' };
      const noDatesComponent = shallow(<CreditCardDetails {...updatedProps} />);
      expect(noDatesComponent.find('span[children="No expiration provided"]').exists()).toBeTruthy();
    });
  });
});
