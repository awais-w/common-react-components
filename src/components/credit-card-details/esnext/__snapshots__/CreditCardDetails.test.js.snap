// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreditCardDetails should match all props snapshot 1`] = `
<div
  className="cmc-credit-card-details"
>
  <i
    className="cmc-icon"
    role="icon"
  >
    credit_card
  </i>
  <span>
    Argos Card
  </span>
  <span>
    XXXX-XXXX-XXXX-1234
  </span>
  <span>
    01/19 - 05/99
  </span>
  <span>
    Mr Test Person
  </span>
</div>
`;

exports[`CreditCardDetails should match required props snapshot 1`] = `
<div
  className="cmc-credit-card-details"
>
  <i
    className="cmc-icon"
    role="icon"
  >
    credit_card
  </i>
  <span>
    XXXX-XXXX-XXXX-1234
  </span>
  <span>
    No expiration provided
  </span>
</div>
`;
